### ISSUE TITLE
ASan heap-user-after-free during node.send_single_many_peers

### ISSUE THREAD

**Issue by rkeene** [nanocurrency/raiblocks#1361]
```
[ RUN      ] node.send_single_many_peers
=================================================================
==26183==ERROR: AddressSanitizer: heap-use-after-free on address 0x60b000118740 at pc 0x000000e9ba3e bp 0x7f00568e9270 sp
 0x7f00568e9268
READ of size 8 at 0x60b000118740 thread T2499 (Announce loop)
```

Use at (Announce loop thread):
```
    #5 0x17300cb in rai::active_transactions::announce_votes(std::unique_lock<std::mutex>&) /home/rkeene/devel/raiblocks/
rai/node/node.cpp:2837
```

Freed at (main thread):
```
    #11 0x17313a2 in rai::active_transactions::stop() /home/rkeene/devel/raiblocks/rai/node/node.cpp:3010
```

Allocated at (Block processing thread):
```
    #9 0x1731dbe in rai::active_transactions::add(std::pair<std::shared_ptr<rai::block>, std::shared_ptr<rai::block> >, std::function<void (std::shared_ptr<rai::block>)> const&) /home/rkeene/devel/raiblocks/rai/node/node.cpp:3042
```

**Comment by rkeene** [comment]
Fix committed to master in [c8245e83d9f8e6839961a5164df5ffefb411c5b7]

**Comment by rkeene** [comment]
Test now passes.
